### PR TITLE
Refactor Morgue Slabs/Trays

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Refactors the morgue code to be a bit more object oriented. Both the morgue and crematorium now inherit from the same supertype, and the crematorium tray is now a subtype of the morgue tray.

Also fixes some small bugs that weren't completely addressed in #17221, which came about in particular due to copy-pasted code.

## Why It's Good For The Game
Morgue code is currently full of duplicated code, and certainly needs a good once-over. This should also ensure that any behavior for moving items in/out of morgue trays, no matter the cause (clicking on the tray, the morgue slab, etc). comes from a single source of truth.

## Changelog
:cl:
fix: Ghosts cannot get pulled into morgue trays when the tray itself is clicked.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/morgue.dm [hunk 1]
```diff
@@ -244,20 +253,22 @@
...
-	anchored = 1.0
-	var/cremating = 0
+	var/cremating = FALSE
 	var/id = 1
-	var/locked = 0
-	var/open_sound = 'sound/items/deconstruct.ogg'
+	dir = SOUTH
+
+/** Overridable method for creating the morgue tray */
+/obj/structure/body_slab/create_tray()
```
Comment (SteelSlayer) [inline code/game/objects/structures/morgue.dm:265]: I think you meant this:
```suggestion
/obj/structure/body_slab/crematorium/create_tray()
```

File: code/game/objects/structures/morgue.dm [hunk 2]
```diff
@@ -17,80 +17,61 @@
...
-	. = ..()
-	. += "[status_descriptors[status]]"
+	add_overlay("morgue_[status]")
+
+/obj/structure/body_slab/ex_act(severity)
 
-/obj/structure/morgue/ex_act(severity)
 	switch(severity)
-		if(1.0)
+		if(1)
```
Comment (hal9000PR) [inline code/game/objects/structures/morgue.dm:61]: there are defines for this in `combat.dm`

File: code/game/objects/structures/morgue.dm [hunk 3]
```diff
@@ -170,40 +135,103 @@
...
+	var/static/status_descriptors = list(
+		EXTENDED_TRAY = "The tray is currently extended.",
+		EMPTY_MORGUE = "The tray is currently empty.",
+		UNREVIVABLE = "The tray contains an unviable body.",
+		REVIVABLE = "The tray contains a body that is responsive to revival techniques.",
+		NOT_BODY = "The tray contains something that is not a body.",
+		GHOST_CONNECTED = "The tray contains a body that might be responsive."
+	)
+
+/obj/structure/body_slab/morgue/proc/update()
```
Comment (hal9000PR) [inline code/game/objects/structures/morgue.dm:160]: you cannot define two procs with the same name on the same type path.  you can override it instead by removing the `/proc/` and CI will stop screaming at you.

**Comment by lewcc** [comment]
Looks like #17171 had a similar idea and I had to merge it in. I'll try to deconflict and fix things up

**Comment by lewcc** [comment]
I'm going to close this for now, I'll probably have to revisit the idea from scratch if I want to ever clean this up more.

**Pull request closed without merging**